src/normalizeCqlExecution.ts

```diff
@@ -2,15 +2,37 @@ import { Uri, window } from 'vscode';
 import { buildParameters } from './buildParameters';
 import { executeCQL } from './executeCql';
 
-export async function normalizeCqlExecution(uri: Uri, mode: 'expression' | 'file') {
-  // Needs a distinction between CQL file and single line
+export async function normalizeCqlExecution(uri: Uri, type: string) {
   const isCqlFile = window.activeTextEditor!.document.fileName.endsWith('.cql');
-
-  if (isCqlFile && mode === 'file') {
-    // should normalize data
-    let operationArgs = buildParameters(uri);
-    executeCQL(operationArgs);
+  let operationArgs;
+  if (isCqlFile && type === "file") {
+    operationArgs = buildParameters(uri, undefined);
+  } else if (isCqlFile && type === "expression") {
+    let cursorPosition = window.activeTextEditor!.selection.active;
+    // For now using parsing the definition here, but ideally should be communicating with the Language Server
+    // Could try something like https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_selectionRange after grabbing the start and end positions of a selection
+    // https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_signatureHelp *Definition Signature???*
+    const definitionStatementRegex = new RegExp('(?<=\\bdefine\\s*"\\s*)([^"]+)(?="\\s*:)');
+    let line = window.activeTextEditor!.document.lineAt(cursorPosition).text;
+    // duplicating for some reason, but for now only a single definition anyways
+    let definitionMatch = line.match(definitionStatementRegex);
+    if (definitionMatch && definitionMatch.length) {
```

Comment:
This is cleaner as an early return; see [here](https://www.youtube.com/watch?v=CFRhGnuXG-4) for a quick video with the rationale.

```suggestion
  if (!definitionMatch || !definitionMatch.length) {
    window.showInformationMessage('Only execution of whole CQL files is currently supported.');
    return;
  }
```